omigami/model.py

```diff
@@ -0,0 +1,92 @@
+# TODO: rename
+import numpy as np
+from typing import Any
+from data_types import RandomState
+from sklearn.base import BaseEstimator, clone
+from sklearn.pipeline import Pipeline
+from sklearn.ensemble import RandomForestClassifier
+
+
+class Estimator:
+    def __init__(self, estimator: Any, random_state: RandomState):
+        self._estimator = estimator
+        self._random_state = random_state
+        self.set_random_state(random_state)
+
+    def fit(self, X, y):
+        self._estimator.fit(X, y)
+        return self
+
+    def predict(self, X):
+        return self._estimator.predict(X)
+
+    @property
+    def feature_importances(self):
+        raise NotImplementedError("feature_importance is not implemented")
+
+    def set_random_state(self, random_state: RandomState):
+        raise NotImplementedError("set_random_state is not implemented")
+
+    def _clone_estimator(self):
+        raise NotImplementedError("_clone_estimator is not implemented")
+
+    def clone(self):
+        estimator_clone = self._clone_estimator()
+        return self.__class__(estimator_clone, self._random_state)
+
+    def train_model(self, X, y):
```

Comment:
I find this method strange why do we need it? Couldn't we always clone during fit?

Reply:
yeah, I was thinking to remove it as well, but cloning during fit is not something I'd do because it would be odd if you are acquainted with scikit-learn.
What I'd do is to replave the two calls to this method with `clone().fit()` and that's it